Duck997 proposal on VLSIGR

Duck997 opened the pull request:
<!-- Please fill the following checkboxes -->
---
A good PR should include the following parts:

- [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [x] Include only your own commits
- [x] Exclude irrelevant files, such as binaries, text files, or dot files
- [x] Provide a clear title
- [x] Provide meaningful descriptions in the PR and the commit logs
- [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [x] Make sure tests/CI pass

yungyuc reviewed (changes requested):
Good progress.  Points to address:

- [x] Shorten the project and repository name.
- [ ] Use 79-char line width.

Review protocol: leave a global comment summarizing the changes and notify for review.  At the site of change, leave inline comments to highlight and explain.

Duck997 commented on project/Duck997/README.rst:
```diff
@@ -1,11 +1,11 @@
 ================
-VLSI Global Router
+VLSIGR
```
Changed from "High-Performance VLSI Global Router with Adaptive Algorithm Design" to "VLSIGR" to shorten the project name as requested in feedback.

Duck997 commented on project/Duck997/README.rst:
```diff
@@ -1,11 +1,11 @@
 =================
 
-**GitHub Repository**: https://github.com/Duck997/VLSI-Router
+**GitHub Repository**: https://github.com/Duck997/VLSIGR
```
Updated repository URL from VLSI-Global-Router to VLSIGR to match the shortened name requirement.

Duck997 commented on project/Duck997/README.rst:
```diff
@@ -232,12 +232,7 @@ Engineering Infrastructure
-     strategies.
-   - **Cross-platform Compatibility**: Ensure compatibility across different
-     environments.
+   Testing: Catch2 (for C++), pytest (for Python).
```
Simplified from detailed testing contents to framework specification, following the standard format used by other proposals as suggested in feedback.

Duck997 commented:
- [x] Shorten the project and repository name.
- [x]  Use 79-char line width.

I have addressed all the feedback points:

✅ **Shortened project and repository name**:
- Project title: "High-Performance VLSI Global Router with Adaptive Algorithm Design" → "VLSIGR"
- Repository name: "VLSI-Global-Router" → "VLSIGR"
- Updated all API references from VLSIRouter to VLSIGR

✅ **Applied 79-character line width**:
- Reformatted all long lines to comply with 79-character limit
- Updated project description, system architecture, API examples, and references

✅ **Fixed testing framework description**:
- Changed from detailed testing contents to framework specification
- Now follows the standard format: "Catch2 or GoogleTest (for C++), pytest (for Python)"

✅ **Updated project list**:
- Modified project/README.rst to reflect the new project name and repository URL

✅ **Fixed reference shortlink**:
- Corrected reStructuredText reference format from numbered list to proper `.. [n]` format
- Now `[3]_` in Layer Assignment section properly links to reference [3] in References section

Sorry for the multiple commits, I'm still new to GitHub workflow.

@yungyuc

Duck997 reviewed (commented):
Fix merge conflict by adding heyvrann's proposal.

Duck997 commented:
I have addressed the issues in the commit "Resolve merge conflict and fix section title":

✅ **Resolved merge conflict**: Fixed the conflict in `project/README.rst` and maintained the VLSIGR project entry.

✅ **Fixed section title**: Changed "Physics and Mathematics" to "Mathematical Foundations" as suggested in the feedback, since the content only covers mathematical concepts without physics.

**Note**: Due to incorrect timezone settings on my VM, the commit timestamps may appear unusual. The commits were made after receiving the feedback, but the system time was not properly configured.

@yungyuc
